### PR TITLE
PR to move to Postgres

### PR DESCRIPTION
**Pull request by maxsocl**
@cjlee112 

In this PR we are moving to Postgresql.

Changes:

- I have added Postgresql setting to local_conf_example.py to set DB username and password.
- max_length of some fields has been increased because of fixture data (e.g. `Lesson.title`) that appeared to be bigger than our old max_length constraints.
- lti/tests.py are changed because of postgresql id's autoincrement feature.

This PR must be merged after https://github.com/cjlee112/socraticqs2/pull/44
After merge of https://github.com/cjlee112/socraticqs2/pull/44  I will do rebase on your master branch and you will see only one new commit to merge - https://github.com/cjlee112/socraticqs2/commit/5943921a564725b674a4edcaee1352a4fdf1e998

@auraz

### CODE REVIEW

**Comment by maxsocl** [comment]
@cjlee112 @auraz PR rebased on master. Plz review it. Thanks.

**Comment by cjlee112** [comment]
@maxsocl Should the PostgreSQL installation docs perhaps touch on security configuration?  As I understand it, once installed on the developer's computer, PostgreSQL will be running all the time as a server.  Hence our PostgreSQL installation instructions should default to block access to all requests except localhost.  I.e. the developer should be able to test Socraticqs2 using the postgres backend, but should not have to worry that installing this opens a new security hole on his computer.

**Comment by cjlee112** [comment]
@maxsocl I can also imagine that some developers may not want to have the PostgreSQL server running all the time on their computer, just to play with the Socraticqs2 code (e.g. this consumes system resources).  Could you add docs indicating how a developer could configure PostgreSQL *not* to start the server by default (i.e. all the time), and how to start / stop the server manually?

**Comment by maxsocl** [comment]
> @maxsocl Should the PostgreSQL installation docs perhaps touch on security configuration? As I understand it, once installed on the developer's computer, PostgreSQL will be running all the time as a server. Hence our PostgreSQL installation instructions should default to block access to all requests except localhost. I.e. the developer should be able to test Socraticqs2 using the postgres backend, but should not have to worry that installing this opens a new security hole on his computer.

In default configuration PostgreSQL is not accepting only localhost connections.
And when we need that postgresql accepts connections for other hosts we need to change configs.

**Comment by maxsocl** [comment]
> @maxsocl I can also imagine that some developers may not want to have the PostgreSQL server running all the time on their computer, just to play with the Socraticqs2 code (e.g. this consumes system resources). Could you add docs indicating how a developer could configure PostgreSQL not to start the server by default (i.e. all the time), and how to start / stop the server manually?

Yes, I can add this off-course.

**Comment by maxsocl** [comment]
> @maxsocl I can also imagine that some developers may not want to have the PostgreSQL server running all the time on their computer, just to play with the Socraticqs2 code (e.g. this consumes system resources). Could you add docs indicating how a developer could configure PostgreSQL not to start the server by default (i.e. all the time), and how to start / stop the server manually?

Done in https://github.com/raccoongang/socraticqs2/commit/2fc7616d947e0b6a33a6bf19d811f168f41f80b4

**Comment by cjlee112** [comment]
@maxsocl New docs describing PostgreSQL setup pushed to our docs server as of now.

**Comment by cjlee112** [comment]
@maxsocl Hmm, we actually need more developer docs updates for this PR.  Look at the section "Database Operations" in devintro.rst, and you'll see various statements that assume sqlite (e.g. that you can "back up the database" by copying the mysite.db file).  Please update all that.

**Comment by cjlee112** [comment]
@maxsocl Actually, what is the equivalent operation for PostgreSQL for backing up or restoring a "snapshot" of the entire database directly (i.e. NOT using django dumpdata / loaddata)?  This is important for all sorts of situations where migrations might go awry.

File: doc/devintro.rst [hunk 1]
```diff
@@ -97,6 +97,104 @@ one of two ways:
...
+  to::
+
+    local   all    postgres    md5
+
+* Finally we need to restart PostgreSQL server::
+
+    sudo service postgresql restart
+
+
+In case when you don't want to have the PostgreSQL server running all the time on your computer, just to play with the Socraticqs2 code (e.g. this consumes system resources), you can start and stop PostgreSQL manually::
```
Comment (cjlee112) [inline doc/devintro.rst]: @maxsocl I'm sad to say these instructions don't work for me.  The sudo service postgres start command gives me OK, but any kind of testing of our app just fails to connect to the database, and indeed sudo service postgres status just reports "down".  There are no error messages in /var/log/postgresql, /var/log/syslog etc.  Totally annoying.  The only way I get it to work is to change "manual" back to "auto" and reboot.  Then the postgres server is indeed up & running.  This was on Mint 17.1.

**Comment by maxsocl** [comment]
@cjlee112 I am sorry. It is my mistake. I need to add to documentation that if developer decided not start server automatically and changed start.conf to manual - he need to use `sudo pg_ctlcluster 9.4 main start` command to start postgres.
In fact there is a couple of way to configure postgres not to start on boot. I think I need to describe all of them and how to start postgres manually in each case.

**Pull request merged**